lib/bolt/plugin/aws.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module Bolt
+  class Plugin
+    class Aws
+      class EC2
+        attr_accessor :client
+        attr_reader :config
+
+        def initialize(config)
+          require 'aws-sdk-ec2'
+          @config = config
+          @logger = Logging.logger[self]
+        end
+
+        def name
+          'aws::ec2'
+        end
+
+        def hooks
+          %w[inventory_targets]
+        end
+
+        def config_client(opts)
+          return client if client
+
+          options = {}
```

Comment:
I think you can shorten this a bit, since we're not doing any checking or munging of the `region` or `profile` keys:
```
opts.transform_keys! &:to_s
```
then just use `opts` instead of `options` for the credentials.

Reply:
To explain some ruby trickery: `!` will modify an object inplace, instead of creating a new one.
`&:` creates a new process that applies the following function to whatever it's operating on. Under the hood it calls `to_proc`, and is similar to doing `opts.transform_keys! { |k| k.to_sym }`

Reply:
Since `opts` might contain keys other than `region`, `profile`, or `credentials`, passing the original hash with these other keys will cause the aws sdk to raise an error.